src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.cs

```diff
@@ -1,69 +1,263 @@
                                                                              s_localizableMessageVisibility,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: MsdnUrl,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, PropertyRule, MultipleRule, VisibilityRule);
 
         public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+        {
+            analysisContext.RegisterSymbolAction(DoAnalysis, SymbolKind.Method);
+        }
+
+        private static void DoAnalysis(SymbolAnalysisContext symbolContext)
+        {
+            var methodSymbol = (IMethodSymbol)symbolContext.Symbol;
+            var location = methodSymbol.OriginalDefinition.Locations.First();
+            var typeSymbol = methodSymbol.ContainingSymbol as ITypeSymbol;
+            var operatorName = methodSymbol.Name;
+            if (typeSymbol != null && (methodSymbol.MethodKind == MethodKind.UserDefinedOperator || methodSymbol.MethodKind == MethodKind.Conversion))
+            {
+                if (IsPropertyExpected(operatorName) && operatorName != OpFalseText)
+                {
+                    // don't report a diagnostic on the `op_False` method because then the user would see two diagnostics for what is really one error
+                    // special-case looking for `IsTrue` instance property
+                    // named properties can't be overloaded so there will only ever be 0 or 1
+                    var property = typeSymbol.GetMembers(IsTrueText).OfType<IPropertySymbol>().SingleOrDefault();
+                    if (property == null || property.Type.SpecialType != SpecialType.System_Boolean)
+                    {
+                        symbolContext.ReportDiagnostic(CreateDiagnostic(PropertyRule, location, AddAlternateText, IsTrueText, operatorName));
+                    }
+                    else if (!property.IsPublic())
+                    {
+                        symbolContext.ReportDiagnostic(CreateDiagnostic(VisibilityRule, property.OriginalDefinition.Locations.First(), FixVisibilityText, IsTrueText, operatorName));
```

Comment:
If you had the helper method `LocationOfSymbol` I suggested above, you could using it here too, `LocationOfSymbol(property)`, and below, `LocationOfSymbol(notPublicMethod)`, which would DRY out the code a little.